src/signature.c

```diff
@@ -450,6 +477,38 @@ gboolean cms_verify(GBytes *content, GBytes *sig, CMS_ContentInfo **cms, X509_ST
 		goto out;
 	}
 
+	/* Optionally use certificate signing timestamp for verification */
+	if (r_context()->config->use_cert_signing_time) {
+		STACK_OF(CMS_SignerInfo) *sinfos;
+		CMS_SignerInfo *si;
+		X509_ATTRIBUTE *xa;
+		ASN1_TYPE *so;
+		X509_VERIFY_PARAM *param = X509_VERIFY_PARAM_new();
```

Comment:
These are missing the corresponding `free`s.

Reply:
Not sure but I assume only param needs to be freed as others are only pointers.